src/main/resources/jupyter/edit-mode.js

```diff
@@ -0,0 +1,337 @@
+            click: () => updateLocalCopyWithRemote()
+        }
+    }
+
+    const lockIssueButtons = {
+        'Run in Playground Mode': {
+            click: () => openPlaygroundMode(),
+            'class': 'btn-primary'
+        },
+        'Make a Copy': {
+            click: () => saveAs()
+        }
+    }
+
+    const noRemoteFileButtons = {
+        'Continue working': {
+            click: () => {},
+            'class': 'btn-primary'
+        }
+    }
+
+    const modeBannerId = "notification_mode"
+    const lockConflictTitle = "File is in use"
+    const syncIssueTitle = "File versions out of sync"
+    const syncIssueBody = "Your version of this file does not match the version in the workspace. What would you like to do?"
+    const syncIssueFatalBody = "This file was either deleted or never was stored with us."
+
+    //TODO URL resolution
+    const leoUrl = ''
+        // const leoUrl = 'http://localhost:8081'
+    const welderUrl = leoUrl + `/proxy/${googleProject}/${clusterName}/welder`
+    const localizeUrl = welderUrl + '/localize'
+        // const checkMetaUrl = welderUrl + '/checkMeta'
+    const checkMetaUrl = welderUrl + '/objects/metadata'
+    const lockUrl = welderUrl + '/lock'
+    const lastLockedTimer = 60000 // in ms, should be 60000 in final PR
+
+    const headers = {
+        'Content-Type': 'application/json',
+        'Access-Control-Allow-Headers': '*'
+    }
+
+    const basePayload = {
+        mode: "no-cors",
+        headers: headers
+    }
+
+    //TODO resolve these links
+    const jupyterBaseUrl = `/notebooks/${googleProject}/${clusterName}`
+    const jupyterContentsAPIUrl = jupyterBaseUrl + "/api/contents/"
+
+    const getLocalPath = () => {
+        return {
+            localObjectPath: Jupyter.notebook.notebook_path
+        }
+    }
+
+    function init() {
+        checkMeta()
+        initSyncMaintainer()
+    }
+
+    async function initSyncMaintainer() {
+        let syncMaintainer = setInterval(() => {
+            checkMeta()
+        }, lastLockedTimer)
+
+        window.onbeforeunload(function() {
+            clearInterval(syncMaintainer)
+        })
+    }
+
+    async function checkMeta() {
+        const payload = {
+            ...basePayload,
+            body: JSON.stringify(getLocalPath()),
+            method: 'POST'
+        }
+
+        return fetch(checkMetaUrl, payload)
+            .then(res => {
+                console.log('in first then of checkmeta', res)
+                if (!res.ok && res.status == 412) {
+                    promptUserWithModal(syncIssueTitle, noRemoteFileButtons, syncIssueFatalBody)
+                } else if (!res.ok && res.status >= 400) {
+                    throw Error(res.statusText)
+                }
+                return res.json()
+            })
+            .then(res => {
+                console.log('res in check meta after .json()', res)
+                handleCheckMetaResp(res) //sets meta state
```

Comment:
optional: if lockedBy != null we could show the locked dialog at this point, rather than trying to acquire it

Reply:
There was a discussion around this, and we determined that it would be best to leave it up to the API to determine the lock status, as its best not to trust client-side logic to determine something that could impact other users. Additionally, it allows looser coupling.

Reply:
I'm not suggesting any client-side logic. The Welder API has a `lastLockedBy` field returned in the metadata response, it was supposed to be null if you can lock - non-null if locked by someone else. If that's changed, we should stop returning it in the response JSON and update the design doc.